### PR TITLE
adding edit functionality to bidder classifications TM-2017

### PR DESCRIPTION
**Pull request by SidelineCory24**
initial testing, working on a better triggering when the edit link is clicked, [BE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/465)

### CODE REVIEW

**Comment by SidelineCory24** [comment]
Still troubleshooting checkmark issue (see comments)

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 1]
```diff
@@ -1,32 +1,131 @@
...
+  clientClassifications: state.clientClassifications,
+  isLoading: state.isLoading,
+});
+
+export const mapDispatchToProps = dispatch => ({
+  getClassifications: () => dispatch(getClassifications()),
+});
+
+// export default Classifications;
+export default connect(mapStateToProps, mapDispatchToProps)(Classifications);
```
Comment (SidelineCory24) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:131]: I'm getting a "never used" error here. I worked on setting up `getClassifications` similarly to `submitBid` by setting it up in the other Checkbox files, as well as the mapDispatch/StatetoProps for the url testing.

Comment (elizabeth-jimenez) [reply src/Components/ProfileDashboard/Classifications/Classifications.jsx:131]: Good job on using another component as an example 🦜 

good use of `mapStateToProps ` and `mapDispatchToProps`!

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 2]
```diff
@@ -1,32 +1,131 @@
...
   clientClassifications: [],
+  isLoading: false,
+  getClassifications: EMPTY_FUNCTION,
 };
 
-export default Classifications;
+const mapStateToProps = state => ({
+  classifications: state.classifications,
+  clientClassifications: state.clientClassifications,
+  isLoading: state.isLoading,
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:123]: I'm not sure these 2 are properly working? i would throw some print statements to confirm.

File: src/actions/classifications.js [hunk 3]
```diff
@@ -46,3 +46,16 @@ export function fetchClassifications() {
       });
   };
 }
+
+export function getClassifications() {
+  return (dispatch) => {
+    batch(() => {
+      dispatch(classificationsIsLoading(true));
+      dispatch(classificationsHasErrored(false));
```
Comment (elizabeth-jimenez) [inline src/actions/classifications.js:54]: these 2 were fine for boilerplate code, but you'll also want to make sure you set up some new ones for this function. A good file to check it `src/actions/glossary.js` as it has some update and get functions

Comment (elizabeth-jimenez) [reply src/actions/classifications.js:54]: this will get into Redux and our reducer files

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 4]
```diff
@@ -1,32 +1,131 @@
...
+    console.log(classificationsInput);
+    const pushClass = classificationsInput;
+    if (!pushClass.includes(c)) {
+      pushClass.push(c);
+    } else {
+      const index = pushClass.indexOf(c);
+      if (index > -1) {
+        pushClass.splice(index, 1);
+      }
+    }
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:35]: we should be able to optimize this to 
```
const index = pushClass.indexOf(c);
if (index > -1) {
  pushClass.splice(index, 1);
}
else {
  pushClass.push(c);
}
```
and even further to
```
const index = pushClass.indexOf(c);
index > -1 ? pushClass.splice(index, 1) : pushClass.push(c);
```
Since we are going to use `index` to splice our array, we can also use it to determine if the item is in the array, which you were already doing on line 32. If we weren't going to use `index` at all, then the `includes` route would have been perfect

Comment (SidelineCory24) [reply src/Components/ProfileDashboard/Classifications/Classifications.jsx:35]: Yes ternary ftw 😅

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 5]
```diff
@@ -37,7 +35,7 @@ const Classifications = props => {
   };
 
   const cancelInput = () => {
-    setUserInput(clientClassifications);
+    setUserInput([...clientClassifications]);
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:38]: I think what you had before should work `setUserInput(clientClassifications);`? We confirmed this morning that setting `userInput` to `clientClassifications` (via `setUserInput`) is not where our ref issue was. 

Our ref issue was in handleInput, like this `const pushClass = clientClassifications;`, just like you had suspected 👍

Comment (SidelineCory24) [reply src/Components/ProfileDashboard/Classifications/Classifications.jsx]: reverted back the changes

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 6]
```diff
@@ -1,32 +1,120 @@
...
+    setUserInput(pushClass);
+  };
+
+  const cancelInput = () => {
+    setUserInput(clientClassifications);
+    setEditView(false);
+  };
+
+  const onSubmit = () => {
+    saveUserClassifications(userInput, userId);
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:43]: We can save a call to `saveUserClassifications` if `userInput` is the same as `clientClassifications`. For the new `updateClassifications` that was discussed today, the implementation would be a little different.

Comment (elizabeth-jimenez) [reply src/Components/ProfileDashboard/Classifications/Classifications.jsx:43]: or even better https://lodash.com/docs/4.17.15#difference

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 7]
```diff
@@ -1,8 +1,10 @@
 import { indexOf } from 'lodash';
+import PropTypes from 'prop-types';
 import CheckBox from '../../CheckBox';
-import { CLASSIFICATIONS, CLIENT_CLASSIFICATIONS } from '../../../Constants/PropTypes';
+import { CLASSIFICATIONS, CLIENT_CLASSIFICATIONS, EMPTY_FUNCTION } from '../../../Constants/PropTypes';
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:4]: might be able to shorten this to `'Constants/PropTypes'`

**Comment by elizabeth-jimenez** [comment]
@SidelineCory24 We'll want to wrap this in a feature flag. It sounds intimidating, but don't worry, it's pretty straightforward. Some recent ones are Scott's available_bidders and my unassigned_bidders. Peep 👀  our open PRs for how they are implemented.

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 8]
```diff
@@ -1,32 +1,127 @@
...
+  const cancelInput = () => {
+    setUserInput(clientClassifications);
+    setEditView(false);
+  };
+
+  const onSubmit = () => {
+    const updateDiff = {
+      insert: difference(userInput, clientClassifications),
+      delete: difference(clientClassifications, userInput),
+    };
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:46]: 😍

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 9]
```diff
@@ -1,32 +1,127 @@
...
+            editMode={!editView}
+            updateClassifications={(h) => handleInput(h)}
+            input={userInput}
+          />
+        </div>
       </div>
+      {
+        !isLoading && !editView &&
+        <div className="section-padded-inner-container small-link-container view-more-link-centered">
+          <button className="unstyled-button" onClick={() => setEditView(true)}>
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:73]: try adding this style to the button `align-items: center` to see if the icon better vertically aligns with the text

File: src/actions/classifications.js [hunk 10]
```diff
@@ -46,3 +49,48 @@ export function fetchClassifications() {
...
+    hasErrored: bool,
+  };
+}
+
+export function updateClassificationsIsLoading(bool) {
+  return {
+    type: 'CLASSIFICATIONS_IS_LOADING',
+    isLoading: bool,
+  };
+}
```
Comment (elizabeth-jimenez) [inline src/actions/classifications.js:65]: These can't have the same name as the one's above.  Rename these, do  a search for the ones above, throughout the project and make sure the newly names ones are in the same reducer files as the ones above (should be in almost all the same files as the ones above, except maybe where they are being called).

Comment (elizabeth-jimenez) [reply src/actions/classifications.js:65]: by "name" i meant the `return.type` 😅

**Comment by elizabeth-jimenez** [comment]
This looks really good! 😍 

A few small things, but mainly just: 
- [ ] updating/implementing those reducers
- [ ]  wrapping this all in a feature flag

File: src/actions/classifications.js [hunk 11]
```diff
@@ -52,14 +52,14 @@ export function fetchClassifications() {
...
-    type: 'CLASSIFICATIONS_HAS_ERRORED',
+    type: 'UPDATE_CLASSIFICATIONS_HAS_ERRORED',
     hasErrored: bool,
   };
 }
 
 export function updateClassificationsIsLoading(bool) {
   return {
-    type: 'CLASSIFICATIONS_IS_LOADING',
+    type: 'UPDATE_CLASSIFICATIONS_IS_LOADING',
```
Comment (elizabeth-jimenez) [inline src/actions/classifications.js]: 🦜

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 12]
```diff
@@ -1,32 +1,131 @@
...
   clientClassifications: [],
+  isLoading: false,
+  updateUserClassifications: EMPTY_FUNCTION,
 };
 
-export default Classifications;
+const mapStateToProps = state => ({
+  classifications: state.classifications,
+  isLoading: state.isLoading,
+});
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:124]: I don't think these are hooked into anything. Test it out by removing them, and isLoading from in the component (not classifications tho, that's being passed in as a prop)

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 13]
```diff
@@ -1,32 +1,127 @@
...
 Classifications.defaultProps = {
   classifications: [],
   clientClassifications: [],
+  updateUserClassifications: EMPTY_FUNCTION,
 };
 
-export default Classifications;
+const mapStateToProps = state => ({
+  classifications: state.classifications,
+});
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:120]: 😅  Sorry, I wasn't super clear. We want to remove the whole mapStateToProps, but keep `classifications` elsewhere in the component the same (cause it is being passed in through props). isLoading wasn't being passed in as a prop, so that's why we could remove it everywhere. You'll also have to update line 127 to `export default connect(null, mapDispatchToProps)(Classifications);`

Comment (SidelineCory24) [reply src/Components/ProfileDashboard/Classifications/Classifications.jsx:120]: Ah okay, got it!

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 14]
```diff
@@ -16,17 +18,17 @@ const CheckboxList = ({ list, clientClassifications }) => (
...
           id={c.code}
           label={c.text}
           small
           value={checked}
           key={c.code}
-          disabled
-          checked={checked}
+          disabled={editMode}
           className="tm-checkbox-disabled-alternate"
+          onChange={(h) => updateClassifications(h)}
```
Comment (mjoyce91) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:31]: Can simplify to:

```jsx
onChange={updateClassifications}
```

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 15]
```diff
@@ -1,32 +1,123 @@
...
+      <div className="usa-grid-full section-padded-inner-container">
+        <div className="usa-width-one-whole">
+          <SectionTitle title="Bidder Classifications" icon="tasks" />
+        </div>
+        <div className="usa-width-one-whole">
+          <CheckboxList
+            list={classifications}
+            id="updates"
+            editMode={!editView}
+            updateClassifications={(h) => handleInput(h)}
```
Comment (mjoyce91) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:68]: Can simplify to:

```jsx
updateClassifications={handleInput}
```

File: src/Components/ProfileDashboard/Classifications/Classifications.jsx [hunk 16]
```diff
@@ -1,32 +1,123 @@
...
+            <button
+              type="button"
+              className="saved-search-form-primary-button"
+              onClick={onSubmit}
+            >Save
+            </button>
+            <button
+              type="button"
+              className="saved-search-form-primary-button"
+              onClick={() => cancelInput()}
```
Comment (mjoyce91) [inline src/Components/ProfileDashboard/Classifications/Classifications.jsx:95]: Can simplify to

```jsx
onClick={cancelInput}
```

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 17]
```diff
@@ -19,33 +19,31 @@ const CheckboxList = ({ list, editMode, updateClassifications,
       checked
       className="tm-checkbox-disabled-alternate"
     />
-    <div className="classifications-badges">
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx]: i removed this `classification-badges` div, cause `classifications-badges` already wraps our new stuff

File: src/sass/_profile.scss [hunk 18]
```diff
@@ -60,28 +60,24 @@
       flex-direction: column;
       flex-grow: 1;
 
-      .classifications-badges {
         .classifications-client-badges {
           padding-bottom: 15px;
-          display: inline;
+          min-width: 100%;
```
Comment (elizabeth-jimenez) [inline src/sass/_profile.scss:65]: this is what was the missing piece 😭

File: src/sass/_profile.scss [hunk 19]
```diff
@@ -60,17 +60,35 @@
...
+          .client-badge-container {
+            display: inline-block;
+            text-align: initial;
+            .client-badge {
+              display: inline;
+            }
+          }
+          .classifications-dropdown {
+            float: right;
```
Comment (elizabeth-jimenez) [inline src/sass/_profile.scss:74]: 🙅

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 20]
```diff
@@ -16,17 +20,42 @@ const CheckboxList = ({ list, clientClassifications }) => (
...
+      const tenDiffFlag = c.text === 'Tenured 4' || c.text === 'Differential Bidders' ? true : '';
+      return (
+        <div className="classifications-client-badges">
+          {tenDiffFlag &&
+            <div className="classifications-dropdown">
+              <Picky
+                placeholder={c.text}
+              />
+            </div>
+          }
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:33]: you can use ? scott's : favorite ? thing in order to render the picky : or the ? text :

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:33]: also, want the picky after the icon

**Comment by SidelineCory24** [comment]
Data is showing in the handle input, but now the client cards & status (green/gray) are not working

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 21]
```diff
@@ -16,18 +20,37 @@ const CheckboxList = ({ list, clientClassifications }) => (
...
+          {/* {tenDiffFlag &&
+            <div className="classifications-dropdown">
+              <Picky
+                placeholder={c.text}
+              />
+            </div>
+          } */}
+          <ClientBadge
+            key={c.te_id}
+            type={c.code}
```
Comment (SidelineCory24) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:44]: Updating type to `type={c}` and updating the props for `type` in ClientBadge fixes the status green/grey issue but the client card badge issue still remains.

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 22]
```diff
@@ -16,18 +19,36 @@ const CheckboxList = ({ list, clientClassifications }) => (
...
     />
     {list.map((c) => {
-      const checked = indexOf(clientClassifications, c.code) > -1;
+      // need to update with te_id
+      let checked = false;
+      input.forEach((item) => {
+        if (item.tp_code === c.code || item.code === c.code) {
+          checked = true;
+        }
+      });
```
Comment (SidelineCory24) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:28]: Will need to update with the `te_id` to handle editing tenured4/diff bidder.

**Comment by SidelineCory24** [comment]
Accordion Styling Testing
1. Added new class for styling `classifications-row`
2. Tested changing `.accordion-inverse` image from the plus/minus -> caret. Tested using image from assets path as well as font-awesome
3. Tested changing the width/overall length of the button, however the size readjusts on opening.
4. Wasn’t always able to get the console styling to take effect in the css file

File: src/sass/_accordion.scss [hunk 23]
```diff
@@ -10,6 +10,7 @@
 
   .usa-accordion-button[aria-expanded=false]:not(.tm-nested-accordion-button) {
     background-image: url('#{$asset-path}assets/img/plus-white.svg');
+    // background-image: url('#{$asset-path}assets/img/angle-down-blue.svg');
```
Comment (elizabeth-jimenez) [inline src/sass/_accordion.scss:13]: good job finding this

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 24]
```diff
@@ -1,48 +1,140 @@
...
+        input.forEach((item) => {
+          if (c.seasons.length === 1 && c.seasons[0].id === item) {
+            checked = true;
+          }
+          if (c.seasons.length > 1) {
+            multiBidSeasonFlag = true;
+            c.seasons.forEach((cs) => {
+              if (cs.id === item) checked = true;
+            });
+          }
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:38]: we should be able to replace line 30-38 with
`  c.seasons.forEach((cs) => {
    if (cs.id === item) checked = true;
  });`

and then line 28 to `let multiBidSeasonFlag = c.seasons.length > 1;`

Comment (SidelineCory24) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:38]: done

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 25]
```diff
@@ -1,48 +1,128 @@
-import { indexOf } from 'lodash';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import FontAwesome from 'react-fontawesome';
+import InteractiveElement from 'Components/InteractiveElement';
+import { Row } from '../../Layout';
 import CheckBox from '../../CheckBox';
-import { CLASSIFICATIONS, CLIENT_CLASSIFICATIONS } from '../../../Constants/PropTypes';
+import ClientBadge from '../ClientBadge';
+import { CLASSIFICATIONS, CLIENT_CLASSIFICATIONS, EMPTY_FUNCTION } from '../../../Constants/PropTypes';
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:8]: from 'Constants/PropTypes';

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 26]
```diff
@@ -1,48 +1,128 @@
...
+                    <ClientBadge
+                      // need to update key/id
+                      // from tenured 4 so onChange/handleInput
+                      // works correctly
+                      key={'tenured4'}
+                      type={c}
+                      id={'tenured4'}
+                      status={checked}
+                      showShortCode={false}
+                      onChange={updateClassifications}
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:49]: 🦜  we can just remove the `onChange` from this `ClientBadge`, that way it'll just function as an aggregate icon

**Comment by elizabeth-jimenez** [comment]
image
i think a padding-left of `40px` would look great!

File: src/Components/BidderPortfolio/ClientBadgeList/ClientBadgeList.jsx [hunk 27]
```diff
@@ -7,13 +7,9 @@ const ClientBadgeList = ({ classifications, clientClassifications }) => (
...
       .slice(0, 4)
       .map((c) => {
-        // need to update with te_id
         let checked = false;
         clientClassifications.forEach((item) => {
-          // if (item.te_id === c.te_id) {
-          if (item.tp_code === c.code || item.code === c.code) {
-            checked = true;
-          }
+          c.seasons.forEach((cs) => { if (cs.id === item) checked = true; });
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/ClientBadgeList/ClientBadgeList.jsx:12]: 🦜

File: src/Components/CheckBox/CheckBox.jsx [hunk 28]
```diff
@@ -1,6 +1,6 @@
 import { Component } from 'react';
 import PropTypes from 'prop-types';
-import { EMPTY_FUNCTION } from '../../Constants/PropTypes';
```
Comment (elizabeth-jimenez) [inline src/Components/CheckBox/CheckBox.jsx:3]: 🧼 👍

File: src/Constants/PropTypes.js [hunk 29]
```diff
@@ -588,14 +588,31 @@ export const HIGHLIGHT_POSITION = PropTypes.shape({
 export { SetType };
 
 export const CLASSIFICATION = PropTypes.shape({
-  id: PropTypes.string,
-  show: PropTypes.string,
+  // id: PropTypes.string,
+  // show: PropTypes.string,
+  // text: PropTypes.string,
+  // may need to add new keys?
```
Comment (elizabeth-jimenez) [inline src/Constants/PropTypes.js:594]: this file needs to be cleaned and updated to handle the new data structure Scott set up. I think something like this:
`{"code":"C","text":"Critical Need Language","seasons":[{"id":85,"season_text":null}]}`

but don't trust me on that 🤞

File: src/Constants/SystemMessages.js [hunk 30]
```diff
@@ -124,3 +124,6 @@ export const ADD_TO_INTERNAL_LIST_ERROR = "We're experiencing an error attemptin
 export const REMOVE_FROM_INTERNAL_LIST_ERROR = "We're experiencing an error attempting to remove this client from the Internal list. Please try again.";
 
 export const COMING_SOON = 'Coming Soon';
+
+export const UPDATE_CLASSIFICATIONS_SUCCESS = "Client's classifications have been updated";
```
Comment (elizabeth-jimenez) [inline src/Constants/SystemMessages.js:128]: "Client's classifications updated" seems more consistent with our other messages

File: src/reducers/classifications/classifications.js [hunk 31]
```diff
@@ -24,3 +24,21 @@ export function classifications(state = [], action) {
...
+}
+
+export function updateClassificationsIsLoading(state = false, action) {
+  switch (action.type) {
+    case 'UPDATE_CLASSIFICATIONS_IS_LOADING':
+      return action.isLoading;
+    default:
+      return state;
+  }
+}
```
Comment (elizabeth-jimenez) [inline src/reducers/classifications/classifications.js:44]: We may or may not want to add something for success. We can discuss this on Monday

File: src/sass/_profile.scss [hunk 32]
```diff
@@ -60,17 +60,40 @@
...
     }
 
     .unstyled-button {
       color: $blue-primary;
       font-size: 1.2em;
     }
 
+    .classifications-checkbox {
+      align-items: center;
+    }
```
Comment (elizabeth-jimenez) [inline src/sass/_profile.scss:95]: i'd look at making the [pencil icon] and font a little bigger

File: src/sass/_profile.scss [hunk 33]
```diff
@@ -60,17 +60,40 @@
...
-      }
+        .classifications-client-badges {
+          padding-bottom: 15px;
+          min-width: 100%;
+          .client-badge-container {
+            display: inline-block;
+            text-align: initial;
+            .client-badge {
+              display: inline;
+            }
```
Comment (elizabeth-jimenez) [inline src/sass/_profile.scss:71]: i'd do a quick audit of the scss and make sure all the styling is still being used. There was a lot of back and forth, so im sure we forgot some stuff. Like the `.client-badge {display: inline; }` no longer seems to do anything

**Comment by elizabeth-jimenez** [comment]
Just a few things left before this is good to go:
- [ ] tests
- [ ] unique id on dropdown
- [ ] discussing if we should add a reducer for success
- [ ] PR comments

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx (hunk 25 above)
Comment (SidelineCory24) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:8]: done

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx (hunk 26 above)
Comment (SidelineCory24) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx]: done

File: src/Constants/PropTypes.js (hunk 29 above)
Comment (SidelineCory24) [reply src/Constants/PropTypes.js]: done

File: src/sass/_profile.scss (hunk 32 above)
Comment (SidelineCory24) [reply src/sass/_profile.scss:95]: done

File: src/sass/_profile.scss (hunk 33 above)
Comment (SidelineCory24) [reply src/sass/_profile.scss:71]: done

Comment (SidelineCory24) [reply src/sass/_profile.scss:71]: done

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 34]
```diff
@@ -1,48 +1,145 @@
...
+          <div className="classifications-client-badges">
+            {multiBidSeasonFlag &&
+            <div>
+              <Row className="usa-grid-full">
+                <div className="usa-grid-full toggle-more-container">
+                  <InteractiveElement className="toggle-more classifications-row" onClick={() => updateShowMore(uniqueShowMore)}>
+                    <ClientBadge
+                      // need to update key/id
+                      // from tenured 4 so onChange/handleInput
+                      // works correctly
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:61]: 🧼

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 35]
```diff
@@ -1,48 +1,141 @@
...
-);
+import ClientBadge from '../ClientBadge';
+
+const CheckboxList = ({ list, editView, updateClassifications,
+  input }) => {
+  const [showMore, setShowMore] = useState({});
+
+  useEffect(() => {
+    // setting up showMore vars for classifications that have a season attached
+    setShowMore(Object.assign({}, ...(list.filter(f => f.seasons.length > 1).map(c => ({ [c.text.split(' ')[0]]: false })))));
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:16]: we may be able to replace some of this with Lodash

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx (hunk 34 above)
Comment (SidelineCory24) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:57]: done

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx (hunk 35 above)
Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:16]: @SidelineCory24 I think we can update this to be `setShowMore(list.reduce(a, c) => { if (c.seasons.length > 1){ a[c.text.split(" ")[0]] = false;} return a;}, {});` hope i got the syntax right 🤞

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 36]
```diff
@@ -1,48 +1,138 @@
...
+  }, []);
+
+  function updateShowMore(uniqShowMore) {
+    const newVal = !showMore[uniqShowMore];
+    setShowMore(prevState => ({ ...prevState, [uniqShowMore]: newVal }));
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:21]: see if this update works `setShowMore({...showMore, [uniqShowMore]: newVal});`

File: src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx [hunk 37]
```diff
@@ -13,7 +13,8 @@ const CheckboxList = ({ list, editView, updateClassifications,
 
   useEffect(() => {
     // setting up showMore vars for classifications that have a season attached
-    setShowMore(Object.assign({}, ...(list.filter(f => f.seasons.length > 1).map(c => ({ [c.text.split(' ')[0]]: false })))));
+    // eslint-disable-next-line no-param-reassign
+    setShowMore(list.reduce((a, c) => { if (c.seasons.length > 1) { a[c.text.split(' ')[0]] = false; } return a; }, {}));
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/CheckboxList/CheckboxList.jsx:17]: @mjoyce91 😻

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 🍥 🍬 Looks really good!

**Comment by elizabeth-jimenez** [comment]
Wait to merge until the BE is completed
